pkg/aggregate/events.go

```diff
@@ -25,54 +25,33 @@ type VersionedEvent[A any] struct {
 }
 
 // EventsQueue stores events.
+// Zero-value is a valid empty queue.
 type EventsQueue[A any] struct {
-	aggregate A
-	version   int
-	queue     []VersionedEvent[A]
+	version int
+	queue   []VersionedEvent[A]
 }
 
-// NewEventsQueue creates a new empty EventsQueue.
-func NewEventsQueue[A any](aggregate A) EventsQueue[A] {
-	return EventsQueue[A]{
-		aggregate: aggregate,
-		version:   0,
-		queue:     []VersionedEvent[A]{},
-	}
-}
-
-// NewEventsQueueFromEvents creates a new EventsQueue from given events.
-// The events are applied to the aggregate.
-// The version is set to the last event's version.
-func NewEventsQueueFromEvents[A any](aggregate A, events []VersionedEvent[A]) (EventsQueue[A], error) {
+// NewEventsQueueFromEvents creates a new EventsQueue with version set to the last event's version.
+func NewEventsQueueFromEvents[A any](events []VersionedEvent[A]) (EventsQueue[A], error) {
```

Comment:
we could have some validation of the event versions here

in general we should aim for a monotonic sequence from 1 to N, correct?

let's turn this into an issue and implement later

Reply:
Good point, although it could be problematic to fix if the events already made it to the storage. 🤔

Reply:
🤔

Reply:
I'll create an issue so that we can think about it later

Reply:
https://github.com/ThreeDotsLabs/esja/issues/4